### PR TITLE
Dump query

### PR DESCRIPTION
**Pull request by yuvaljacoby**

### CODE REVIEW

File: maraboupy/MarabouCore.cpp [hunk 1]
```diff
@@ -103,7 +103,9 @@ void createInputQuery(InputQuery &inputQuery, std::string networkFilePath, std::
     printf( "Property: None\n" );
 }
 
-std::pair<std::map<int, double>, Statistics> solve(InputQuery inputQuery, std::string redirect="", unsigned timeout=0){
+/* The default parameters here are just for readability, you should specify
+ * them in the to make them work*/
```
Comment (guykatzz) [inline maraboupy/MarabouCore.cpp]: Sentence is broken

File: maraboupy/MarabouCore.cpp [hunk 2]
```diff
@@ -112,14 +114,14 @@ std::pair<std::map<int, double>, Statistics> solve(InputQuery inputQuery, std::s
     if(redirect.length()>0)
         output=redirectOutputToFile(redirect);
     try{
-        Engine engine;
-        if(!engine.processInputQuery(inputQuery)) return std::make_pair(ret, *(engine.getStatistics()));
+        Engine *engine = new Engine(verbosity);
```
Comment (guykatzz) [inline maraboupy/MarabouCore.cpp]: if you allocate dynamically, you have to delete() (in every possible branch of the execution).

File: maraboupy/MarabouCore.cpp (hunk 1 above)
Comment (yuvaljacoby) [reply maraboupy/MarabouCore.cpp]: That is to keep it less then 80 characters.
We don't want that?

**Comment by yuvaljacoby** [comment]
Tried to create a test for maraboupy to make sure no memory leaks.
Had trouble with the compilation, I think I will try to simplify the compilation process (cmake) and then get back to write the test, anyway the python changes here are small but are suppose to save time for new users using the python 


A bit more about the compilation difficulties  I had - 
I needed to extend the MockEngine & IEngine (to have solve method) .
Then tried to change the MarabouCore to work with IEngine and then stuff started to break in MarabouCore and in SmtCore tests

**Comment by yuvaljacoby** [comment]
@guykatzz  Can I merge this?

**Comment by guykatzz** [comment]
Yes

**Pull request merged**